courses/models.py

```diff
@@ -273,12 +272,7 @@ def has_exam(self):
         """
         Check if the user is authorized for an exam for a course run
         """
-        return (
-            self.edx_course_key and
-            self.course and
-            not is_blank(self.course.exam_module) and
-            not is_blank(self.course.program.exam_series_code)
-        )
+        return self.course.exam_runs.filter(date_last_eligible__gt=datetime.now(pytz.utc)).exists()
```

Comment:
Generally, learners need to register 24 hours before the last exam, so `gt` is right here I think. It's somewhat rough timing anyway. @pdpinch does that make sense?